compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -85,17 +85,21 @@ pub(super) fn compare_impl_method<'tcx>(
 
 /// This function is best explained by example. Consider a trait:
 ///
-///     trait Trait<'t, T> {
-///         // `trait_m`
-///         fn method<'a, M>(t: &'t T, m: &'a M) -> Self;
-///     }
+/// ```rust,ignore (pseudo-Rust)
+/// trait Trait<'t, T> {
+///     // `trait_m`
+///     fn method<'a, M>(t: &'t T, m: &'a M) -> Self;
+/// }
+/// ```
 ///
 /// And an impl:
 ///
-///     impl<'i, 'j, U> Trait<'j, &'i U> for Foo {
-///          // `impl_m`
-///          fn method<'b, N>(t: &'j &'i U, m: &'b N) -> Foo;
-///     }
+/// ```rust,ignore (pseudo-Rust)
```

Comment:
If you move the "and an impl" comment above you can merge the code blocks and also remove the ignore from here (you'll need to declare a  `struct Foo;`)